Zend/zend_vm_def.h

```diff
@@ -6370,6 +6370,69 @@ ZEND_VM_C_LABEL(isset_object_finish):
 	ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 }
 
+ZEND_VM_HANDLER(199, ZEND_ARRAY_KEY_EXISTS, CV|TMPVAR|CONST, CV|TMPVAR|CONST)
+{
+	USE_OPLINE
+
+	zend_free_op free_op1, free_op2;
+	zval *key, *subject;
+	HashTable* ht;
+	int result;
+
+	SAVE_OPLINE();
+
+	key = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
+	subject = GET_OP2_ZVAL_PTR_UNDEF(BP_VAR_R);
+
+ZEND_VM_C_LABEL(try_again_subject):
+	if (EXPECTED(Z_TYPE_P(subject) == IS_ARRAY)) {
+		ht = Z_ARRVAL_P(subject);
+	} else if (UNEXPECTED(Z_TYPE_P(subject) == IS_OBJECT)) {
+		ht = Z_OBJPROP_P(subject);
+	} else if (Z_ISREF_P(subject)) {
+		subject = Z_REFVAL_P(subject);
+		ZEND_VM_C_GOTO(try_again_subject);
+	} else {
+		if (OP1_TYPE == IS_CV && UNEXPECTED(Z_TYPE_P(key) == IS_UNDEF)) {
+			ZVAL_UNDEFINED_OP1();
+		}
+		if (OP2_TYPE == IS_CV && UNEXPECTED(Z_TYPE_INFO_P(subject) == IS_UNDEF)) {
+			subject = ZVAL_UNDEFINED_OP2();
+		}
+		zend_internal_type_error(EX_USES_STRICT_TYPES(), "array_key_exists() expects parameter 2 to be array, %s given", zend_get_type_by_const(Z_TYPE_P(subject)));
+		FREE_OP2();
+		FREE_OP1();
+		ZVAL_NULL(EX_VAR(opline->result.var));
+		ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
+	}
+
+ZEND_VM_C_LABEL(try_again_key):
+	if (EXPECTED(Z_TYPE_P(key) == IS_STRING)) {
+		result = zend_symtable_exists_ind(ht, Z_STR_P(key));
+	} else if (EXPECTED(Z_TYPE_P(key) == IS_LONG)) {
+		result = zend_hash_index_exists(ht, Z_LVAL_P(key));
+	} else if (UNEXPECTED(Z_TYPE_P(key) == IS_NULL)) {
+		result = zend_hash_exists_ind(ht, ZSTR_EMPTY_ALLOC());
+	} else if (Z_TYPE_P(key) <= IS_NULL) {
+		if (OP1_TYPE == IS_CV && UNEXPECTED(Z_TYPE_P(key) == IS_UNDEF)) {
+			ZVAL_UNDEFINED_OP1();
```

Comment:
Must be OP2_TYPE  ans  ZVAL_UNDEFINED_OP2.

Reply:
It seems, you changed OP1->OP2 in wrong place. I'll take care about this and merge.

Reply:
Oh, my bad. Everything is fine.

Reply:
Key is OP1 and haystack is OP2 so it should be fine.
https://github.com/php/php-src/compare/9d438180daf644b9937ba4177c27b7e9e95e05cf..f4731326d340d9f67d608201a95ef182af223db2#diff-3054389ad750ce9a9f5895cd6d27800fR6400